hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkUtils.java

```diff
@@ -224,19 +220,37 @@ public ExecRow call(ExecRow row) throws Exception {
         SpliceBaseOperation operation = (SpliceBaseOperation) serverSideResultSet;
         DataSetProcessor dsp = EngineDriver.driver().processorFactory().distributedProcessor();
         DataSet<ExecRow> spliceDataSet = operation.getResultDataSet(dsp);
-        if(spliceDataSet instanceof SparkDataSet) {
-            JavaRDD<ExecRow> rdd = ((SparkDataSet)spliceDataSet).rdd;
-            final ResultColumnDescriptor[] columns = serverSideResultSet.getResultDescription().getColumnInfo();
-
-            // Generate the schema based on the ResultColumnDescriptors
-            List<StructField> fields = new ArrayList<>();
+        
+        final ResultColumnDescriptor[] columns = serverSideResultSet.getResultDescription().getColumnInfo();
+        // Generate the schema based on the ResultColumnDescriptors
+        List<StructField> fields = new ArrayList<>();
+        for (ResultColumnDescriptor column : columns) {
+            fields.add(column.getStructField());
+        }
+        if( fields.stream().map( f -> f.name().toUpperCase() ).distinct().count() != fields.size() ) {  // has duplicate names
```

Comment:
Do we need toUpperCase() here? It does not seem consistent with the subsequent HashSet logic, which honors case sensitivity.

Reply:
I think I could leave that and change the HashSet uses to `used.contains(name.toUpperCase())` and `used.add(name.toUpperCase())`.  I would also leave the case of the name passed to StructField unchanged.

Reply:
Thanks @jpanko1 ! We could have case sensitive column names. For example:
```
splice> create table t1 ("a1" int, "A1" int); 
0 rows inserted/updated/deleted
ELAPSED TIME = 19 milliseconds
splice> select * from t1;
a1         |A1         
-----------------------

0 rows selected
ELAPSED TIME = 28 milliseconds
```
After the binding phase, all the regular column names should have already been in the UPPER case except for the case sensitive column names. With that consideration, should we remove the toUpperCase() call?

Reply:
I see, thanks Yi!  I've removed `toUpperCase()`.
I also checked about case sensitivity in Spark and found that it can be set either way with a param: `spark.sqlContext.setConf("spark.sql.caseSensitive", "true")`.

Reply:
Thanks @jpanko1 for checking!